Userland/Libraries/LibArchive/Tar.cpp

```diff
@@ -35,4 +35,21 @@ bool TarFileHeader::content_is_like_extended_header() const
     return type_flag() == TarFileType::ExtendedHeader || type_flag() == TarFileType::GlobalExtendedHeader;
 }
 
+void TarFileHeader::set_filename_and_prefix(StringView filename)
+{
+    // FIXME: Add support for extended tar headers for longer filenames.
+    VERIFY(filename.length() <= sizeof(m_filename) + sizeof(m_prefix));
+
+    if (filename.length() <= sizeof(m_filename)) {
+        set_prefix(""sv);
+        set_filename(filename);
+        return;
+    }
+
+    Optional<size_t> slash = filename.find('/', filename.length() - sizeof(m_filename));
+
+    VERIFY(slash.has_value());
+    set_prefix(filename.substring_view(0, slash.value() + 1));
+    set_filename(filename.substring_view(slash.value() + 1));
+}
 }
```

Comment:
Please always add a newline before closing namespaces

Reply:
that fails our clang-format

Reply:
It most definitely doesn't, before you made this change there was an empty line.

Reply:
You're right. my clang-format doesn't care. my git-clang-format however, removes the newline.